src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,170 @@
+
 package core.basesyntax;
 
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private static final String INDEX_FOR_EXCEPTION = "Index: ";
+    private static final String SIZE_FOR_EXCEPTION = ", Size: ";
+    private static final int NUMBER_TO_SUBTRACT = 1;
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
+    private static class Node<T> {
+        private T value;
+        private Node<T> next;
+
+        Node(T value) {
+            this.value = value;
+        }
+    }
```

Comment:
All private inner classes and methods should be after the public ones (recommended)